### ISSUE TITLE
Add "ansi" to `unsupported_term`

### ISSUE THREAD

**Issue by bdarnell** [antirez/linenoise#109]
In my admittedly idiosyncratic setup (I cargo-culted it from somewhere so there's at least one other person doing this, but probably not many), I use emacs shells with `TERM=ansi`, because my emacs configuration supports basic ANSI escapes (color etc) but not fancier things like cursor positioning. In `linenoise` this should be treated as a dumb/unsupported terminal.

**Comment by Sonophoto** [comment]
Isn't the truth that your ANSI implementation is incomplete? 
When the user sets `TERM=ansi` that says to me that I can count on using linenoise functions that use ANSI sequences to do cursor control like `linenoiseClearScreen()`

**Comment by bdarnell** [comment]
Could be; as I said I cargo-culted this and I'm not sure what's technically correct. All I know is that `TERM=ansi` (intended to mean color-coded escapes without certain other features) has worked for me for years and that e.g. applications using `libreadline` act as I expect. 

After a little more research, the problem is in `getCursorPosition`: `linenoise` sends `"\x1b[6n"` and the terminal never responds with the cursor position, resulting in deadlock. Is there some way to negotiate the use of this feature? (this is after an `ioctl()` fails)

**Comment by Sonophoto** [comment]
I'm not aware that `TERM=ansi` has some limited meaning at all, I think it means `TERM=ansi`

This is ANSI standard X3.64 (digested). To be technically correct, anything that claims to be this terminal must support the following interface (completely) otherwise the statement `TERM=ansi` is false anyway, right?: 

http://www.everything2.com/title/ANSI+X3.64

**Comment by bdarnell** [comment]
Yeah, `TERM=ansi` may not be the best name for a terminal with these capabilities. But then what is? `dumb` or `emacs` will work with `linenoise` but will disable color in most other applications that respect `terminfo`. I can trawl through `terminfo` for the right variant (`ecma+color` or `ecma+sgr` look like reasonable choices: ECMA-48 is the successor to ANSI X3.64), but it won't be on `linenoise`'s blacklist.

Most escape sequences are one-way - if you use the color codes on a dumb terminal, all you're going to do is blast garbage all over the screen (as an emacs shell user, I'm kind of used to this). The `[6n` sequence used in `getCursorPosition` is different because it expects a response from the terminal, and at least in the case of `linenoise` it blocks the application waiting for that response. This is a more severe failure mode and argues for being more conservative in the use of this feature, or at least using a timeout while waiting.

**Comment by Sonophoto** [comment]
I'm sorry, I don't know what to tell you. It sounds like you know quite a bit about it :-)
